sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/DotNetSensorTest.java

```diff
@@ -164,10 +155,64 @@ public void whenReportsArePresentThereAreNoWarnings() {
   }
 
   @Test
-  public void noAnalysisIfNoFilesDetected() {
+  public void whenThereAreBothMainAndTestFiles_doNotLog() {
+    addMainFileToFs();
+    addTestFileToFs();
+    // add roslyn reports to avoid warnings in the logs for this test
+    addRoslynReports();
+
+    sensor.execute(tester);
+
+    assertThat(logTester.logs(LoggerLevel.DEBUG)).isEmpty();
+    assertThat(logTester.logs(LoggerLevel.WARN)).isEmpty();
+  }
+
+  @Test
+  public void whenThereAreOnlyTestFiles_logWarning() {
+    addTestFileToFs();
+
+    sensor.execute(tester);
+
+    assertThat(logTester.logs(LoggerLevel.DEBUG)).isEmpty();
+    assertThat(logTester.logs(LoggerLevel.WARN))
+      .containsExactly("This solution contains only TEST files. Because there are no MAIN files, this sensor will be skipped. Your SonarQube/SonarCloud project will not have results for LANG_KEY files.");
+  }
+
+  @Test
+  public void whenThereAreNoFiles_logDebug() {
     sensor.execute(tester);
 
-    assertThat(logTester.logs(LoggerLevel.DEBUG)).hasSize(1);
-    assertThat(logTester.logs(LoggerLevel.DEBUG).get(0)).isEqualTo("No files to analyze. Skip Sensor.");
+    assertThat(logTester.logs(LoggerLevel.WARN)).isEmpty();
+    assertThat(logTester.logs(LoggerLevel.DEBUG))
+      .containsExactly("No files to analyze. Skip Sensor.");
+  }
+
+  private void addMainFileToFs() {
+    DefaultInputFile mainFile = new TestInputFileBuilder("mod", "file.language")
+      .setLanguage(LANG_KEY)
+      .setType(Type.MAIN)
+      .build();
+    addFileToFs(mainFile);
+  }
+
+  private void addTestFileToFs() {
+    DefaultInputFile testFile = new TestInputFileBuilder("mod", "SomeTest.language")
+      .setLanguage(LANG_KEY)
+      .setType(Type.TEST)
+      .build();
+    addFileToFs(testFile);
+  }
+
+  private void addFileToFs(InputFile inputFile) {
```

Comment:
By adding type and name arguments here, we can remove the duplicated logic from the two methods above. It takes 5 steps to build the file.